platform/server/detect.py

```diff
@@ -222,6 +222,13 @@ def configure(env):
     if not env["builtin_pcre2"]:
         env.ParseConfig("pkg-config libpcre2-32 --cflags --libs")
 
+    # Embree is only compatible with x86_64. Yet another unreliable hack that will break
+    # cross-compilation, this will really need to be handle better. Thankfully only affects
+    # people who disable builtin_embree (likely distro packagers).
+    if env["tools"] and not env["builtin_embree"] and (is64 and platform.machine() == "x86_64"):
```

Comment:
For the reference, this can break when building headless on macOS if users pass `builtin_embree=no` and they don't have embree3 in their system path.

But that's not a supported configuration and the same applies to all `builtin_*` options here which are intended only for Linux distro packaging, so I'm not making this specific check more thorough than the rest.